server/src/main/java/io/druid/query/lookup/LookupModule.java

```diff
@@ -185,30 +187,34 @@ public LookupResourceListenerAnnouncer(
 class LookupListeningAnnouncerConfig extends ListeningAnnouncerConfig
 {
   public static final String DEFAULT_TIER = "__default";
+  private final DataSourceTaskIdHolder dataSourceTaskIdHolder;
   @JsonProperty("lookupTier")
   private String lookupTier = null;
+  @JsonProperty("lookupTierIsDatasource")
+  private boolean lookupTierIsDatasource = false;
 
   @JsonCreator
-  public static LookupListeningAnnouncerConfig createLookupListeningAnnouncerConfig(
+  public LookupListeningAnnouncerConfig(
       @JacksonInject ZkPathsConfig zkPathsConfig,
-      @JsonProperty("lookupTier") String lookupTier
```

Comment:
no, this actually was not effective as the field annotation takes precedence when using the json configurator.